packages/core/src/util/firmware.ts

```diff
@@ -9,6 +9,28 @@ export interface Firmware {
 	firmwareTarget?: number;
 }
 
+/**
+ * Guess the firmware format based on filename and firmware buffer
+ *
+ * @param filename The firmware filename
+ * @param firmware The firmware file data buffer
+ */
+export function guessFirmwareFormat(
+	filename: string,
+	firmware: Buffer,
+): FirmwareFileFormat {
+	if (
+		(filename.endsWith(".exe") || filename.endsWith(".ex_")) &&
+		firmware.includes(Buffer.from("Aeon Labs", "utf8"))
+	) {
+		return "aeotec";
+	} else if (/\.(hex|ota|otz)$/.test(filename)) {
+		return filename.slice(-3) as FirmwareFileFormat;
+	} else {
```

Comment:
Should we throw a separate error to let users know that `.hec` is not supported?

Reply:
Humm nope, it's invalid like the others for now until we don't support it

Reply:
I mean saying `.hec` is not (and probably won't be supported) is clearer than "we don't know what this is, maybe we could support it".